### ISSUE TITLE
Using Select Component when child object is positioned at [0,0,0] (AR)

### ISSUE THREAD

**Issue by AdamRamberg** [react-spring/react-xr#6]
If an object is positioned at the origin of the scene (0,0,0) and that object is wrapped by a [Select](https://github.com/react-spring/react-xr/blob/299273209decc8d6c49351b2e4df799b475f0380/src/Interactions.tsx#L30) component, the controller’s [hovering state](https://github.com/react-spring/react-xr/blob/299273209decc8d6c49351b2e4df799b475f0380/src/XR.tsx#L92) for that object is set to true at start before any [event handler is hooked up](https://github.com/react-spring/react-xr/blob/299273209decc8d6c49351b2e4df799b475f0380/src/Interactions.tsx#L48). This results in that the user has to click besides the object once before the Select component works. This behavior can be seen in the [demo AR scene](https://codesandbox.io/s/react-xr-simple-ar-demo-8w8hm?file=/src/App.tsx:724-732) if you position the Box at [0,0,0]. 

The reason for this behavior is that controller’s initial position is [0,0,0], and therefore the controller’s ray intersects with the object at the origin at start. 

My suggestion is to add a check that `inputSource` is defined before adding an object to the hovering state of a controller. 

```
            if (it.inputSource && !hovering.has(eventObject) && handlers.onHover.has(eventObject)) {
              hovering.add(eventObject)
              handlers.onHover.get(eventObject)?.({ controller: it, intersection })
            }
```
At: https://github.com/react-spring/react-xr/blob/299273209decc8d6c49351b2e4df799b475f0380/src/XR.tsx#L91

This works since `inputSource` gets defined in first the controller's [connected event handler](https://github.com/react-spring/react-xr/blob/299273209decc8d6c49351b2e4df799b475f0380/src/XR.tsx#L50). When running in AR mode (haven't tested in VR mode) `inputSource` is only defined when the user is touching the screen (not sure if this is the intended behaviour, if not my fix above might not be the best solution). 

Tested this on Android in Chrome 84. 

Let me know if this is a desired solution. If it is I can make a PR.

**Comment by sniok** [comment]
Good catch! Your solution sounds good, we shouldn't trigger events when 'controller' is not loaded yet

**Comment by sniok** [comment]
#7 is published in version 1.0.9
